torch/csrc/distributed/c10d/TCPStoreBackend.cpp

```diff
@@ -228,11 +229,20 @@ void TCPStoreMasterDaemon::clearSocketWaitState(int socket) {
 
 // query communicates with the worker. The format
 // of the query is as follows:
-// type of query | size of arg1 | arg1 | size of arg2 | arg2 | ...
+// queryMagicNumber | type of query | size of arg1 | arg1 | size of arg2
+// | arg2 | ...
 // or, in the case of wait
-// type of query | number of args | size of arg1 | arg1 | ...
+// queryMagicNumber | type of query | number of args | size of arg1
+// | arg1 | ...
 void TCPStoreMasterDaemon::query(int socket) {
   QueryType qt;
+  uint32_t qPrefix;
+  tcputil::recvBytes<uint32_t>(socket, &qPrefix, 1);
+  if (qPrefix != detail::queryMagicNumber) {
+    C10D_DEBUG("torch.distributed.TCPStore: a miscellaneous query is detected.");
```

Comment:
I believe it would be more consistent to do the same as [here](https://github.com/pytorch/pytorch/pull/107607/files#diff-f48445103ab88f3d718000ff47ba51c8eed17fdba7b3007b39ecd2914bbaecc3R279). Instead of C10D_DEBUG and then return, using:

`TORCH_CHECK(false, "Unexpected query magic number");`

Reply:
Fixed. Thanks for the comment, @wenjianhn.